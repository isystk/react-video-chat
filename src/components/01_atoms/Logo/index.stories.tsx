import { Meta, Story } from '@storybook/react'
import React from 'react'
import Logo from './index'

export default {
  title: '01_atoms/Logo',
  component: Logo,
} as Meta

const Template: Story = (props) => {
  return <Logo name="sample" />
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
