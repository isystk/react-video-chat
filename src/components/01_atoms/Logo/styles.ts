import { css, keyframes } from '@emotion/css'

const useStyles = () => css`{
}`

export { useStyles }
