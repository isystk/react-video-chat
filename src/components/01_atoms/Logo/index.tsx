import React, { FC } from 'react'
import { ContainerProps, WithChildren } from 'types'
import { Typography } from '@material-ui/core'
import { connect } from '@/components/hoc'
import { useStyles } from './styles'

/** LogoProps Props */
export type LogoProps = WithChildren & { name }
/** Presenter Props */
export type PresenterProps = LogoProps & { classes }

/** Presenter Component */
const LogoPresenter: FC<PresenterProps> = ({ name, classes }) => (
  <>
    <Typography variant="h6" component="div">
      {name}
    </Typography>
  </>
)

/** Container Component */
const LogoContainer: React.FC<ContainerProps<LogoProps, PresenterProps>> = ({
  presenter,
  children,
  ...props
}) => {
  const classes = useStyles()
  return presenter({
    children,
    classes,
    ...props,
  })
}

export default connect<LogoProps, PresenterProps>(
  'Logo',
  LogoPresenter,
  LogoContainer
)
