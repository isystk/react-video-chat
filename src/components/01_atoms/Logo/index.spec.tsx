import React from 'react'
import renderer from 'react-test-renderer'
import Logo from './index'
import '@testing-library/jest-dom/extend-expect'

describe('Logo', () => {
  it('Match Snapshot', () => {
    const component = renderer.create(<Logo name="sample" />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
