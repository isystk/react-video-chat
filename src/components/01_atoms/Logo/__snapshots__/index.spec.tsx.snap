// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Match Snapshot 1`] = `
<div
  className="MuiTypography-root MuiTypography-h6"
>
  sample
</div>
`;
